packages/medusa/src/services/csv-parser.ts

```diff
@@ -0,0 +1,95 @@
+import { AwilixContainer } from "awilix"
+import {
+  AbstractParser,
+  LineContext,
+  ParserOptions,
+} from "../interfaces/abstract-parser"
+import { CsvSchema, ICsvParser } from "../interfaces/csv-parser"
+import Papa, { ParseConfig } from "papaparse"
+import fs from "fs"
+
+class CsvParser<
+    TSchema extends CsvSchema = CsvSchema,
+    TParserResult = any,
+    TOutputResult = any
+  >
+  extends AbstractParser<TSchema, TParserResult, TOutputResult>
+  implements ICsvParser<TParserResult>
+{
+  $$delimiter = ";"
+
+  constructor(
+    protected readonly container: AwilixContainer,
+    schema: TSchema,
+    delimiter?: string
+  ) {
+    super(schema)
+    if (delimiter) {
+      this.$$delimiter = delimiter
+    }
+  }
+
+  public async parse(
+    path: string,
+    options: ParserOptions = {
+      dynamicTyping: true,
+      header: true,
+    }
+  ): Promise<TParserResult[]> {
+    const fileStream = fs.createReadStream(path)
+    const csvStream = Papa.parse(Papa.NODE_STREAM_INPUT, {
```

Comment:
question: why do we spread and cast the options here? Without the update of the type as suggested before, the ParserOptions is ParseConfig